llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -4056,6 +4056,25 @@ Instruction *InstCombinerImpl::visitOr(BinaryOperator &I) {
   return nullptr;
 }
 
+static Instruction *foldXorToOr(BinaryOperator &I,
+                                InstCombiner::BuilderTy &Builder) {
+  assert(I.getOpcode() == Instruction::Xor);
+  Value *A, *B, *C, *D, *E;
+
+  // ((select C, A, B) | E) ^ D) -> (select C, A ^ D, B ^ D) | E)
+  if (match(&I,
```

Comment:
move this condition to the `InstCombinerImpl::visitXor` function.
I can help with the condition, but I let you think.
further hints:
I think I should refer to @dtcxzyw to double-check my suggestions:
- I think the issue was about matching constants in place of `A`, `B`, and `D`, so It's better to use `m_APInt` instead.

Reply:
Will this check be profitable if the `or` has more than one use? 🤔

Reply:
> Will this check be profitable if the `or` has more than one use? 🤔

Probably not.